pulser-core/pulser/sequence/sequence.py

```diff
@@ -1632,29 +1638,25 @@ def draw(
     def _plot(self, **draw_options: bool) -> tuple[Figure | None, Figure]:
         return draw_sequence(self, **draw_options)
 
-    def _add_slm_pulse(self, pulse: Pulse, channel: str) -> None:
+    def _add_slm_pulse(self, pulse: Pulse) -> None:
         if (
-            self._in_ising
-            and self._slm_mask_dmm
-            and self.declared_channels[channel].addressing == "Global"
-            and isinstance(self.declared_channels[channel], DMM)
-            and not self._schedule[self._slm_mask_dmm]._has_pulse_slot
-            and not _ChannelSchedule.is_detuned_delay(pulse)
+            not _ChannelSchedule.is_detuned_delay(pulse)
+            and self._slm_mask_dmm is not None
         ):
             bottom_detuning = cast(
                 DMM, self.declared_channels[self._slm_mask_dmm]
             ).bottom_detuning
-            min_det = -10 * np.max(pulse.detuning.samples)
+            min_det = -10 * np.max(pulse.amplitude.samples)
             min_det = (
                 bottom_detuning
                 if (bottom_detuning and min_det < bottom_detuning)
                 else min_det
             )
-            self._schedule[self._slm_mask_dmm]._has_pulse_slot = True
+            self._schedule[
+                self._slm_mask_dmm
+            ]._waiting_for_first_pulse = False  # type: ignore
```

Comment:
Tired of all the casting? 😅 
Since you are casting further down anyway, why not do it once and store it? Like this
```python
dmm_schedule = cast(_DMMSchedule, self._schedule[self._slm_mask_dmm])
```
EDIT: Nevermind, it was a different method.

Reply:
No not tired haha I tried the casting but it was but as I feared by putting `_waiting_for_first_pulse` in only `_ChannelSchedule`, althought it makes sense I don't know how to handle modifying this attribute in `self._schedule[self._slm_mask_dmm]` (`cast` creates a new object from what I have observed)

Reply:
That's odd, `cast` should have no effect at runtime, much less create a new object. I tried the `cast`, `mypy` doesn't complain and no test failed. Should I be looking out for anything else?

Reply:
Okay perhaps there was another issue at the time then 😅